### PR TITLE
mysqlSink(cdc): Fix RemoveTable blocking infinitely

### LINKED ISSUE #5847: mysqlSink can deadlock when executing DML results in errors
```
goroutine 765 [chan receive]:
github.com/pingcap/tiflow/cdc/sink/mysql.(*mysqlSinkWorker).cleanup(0xc001c60580)
	/go/src/github.com/pingcap/tiflow/cdc/sink/mysql/mysql_worker.go:161 +0x58
github.com/pingcap/tiflow/cdc/sink/mysql.(*mysqlSink).createSinkWorkers.func1()
	/go/src/github.com/pingcap/tiflow/cdc/sink/mysql/mysql.go:427 +0x347
created by github.com/pingcap/tiflow/cdc/sink/mysql.(*mysqlSink).createSinkWorkers
	/go/src/github.com/pingcap/tiflow/cdc/sink/mysql/mysql.go:418 +0x207

goroutine 940 [select]:
github.com/pingcap/tiflow/cdc/sink/mysql.(*mysqlSink).notifyAndWaitExec(0xc00294c100, {0x79c0528, 0xc001c60340})
	/go/src/github.com/pingcap/tiflow/cdc/sink/mysql/mysql.go:454 +0x3e5
github.com/pingcap/tiflow/cdc/sink/mysql.(*mysqlSink).dispatchAndExecTxns(0xc00294c100, {0x79c0528, 0xc001c60340}, 0x0?)
	/go/src/github.com/pingcap/tiflow/cdc/sink/mysql/mysql.go:498 +0x2b3
github.com/pingcap/tiflow/cdc/sink/mysql.(*mysqlSink).flushRowChangedEvents(0xc00294c100, {0x79c0528, 0xc001c60340})
	/go/src/github.com/pingcap/tiflow/cdc/sink/mysql/mysql.go:274 +0x3d2
created by github.com/pingcap/tiflow/cdc/sink/mysql.NewMySQLSink
	/go/src/github.com/pingcap/tiflow/cdc/sink/mysql/mysql.go:212 +0x22bc
```

MySQL sink fails to handle errors correctly, leading to deadlocks.

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve?

Issue Number: close #5847

### What is changed and how it works?
- Adjusted the error handling logic of `mysqlSink`.

#### Tests 

 - Unit test
 - Integration test

##### Will it cause performance regression or break compatibility?
No.

##### Do you need to update user documentation, design documentation or monitoring documentation?
No.

### Release note 

```release-note
None.
```

### CODE REVIEW

File: cdc/sink/mysql/mysql.go [hunk 1]
```diff
@@ -431,6 +443,9 @@ func (s *mysqlSink) createSinkWorkers(ctx context.Context) error {
 }
 
 func (s *mysqlSink) notifyAndWaitExec(ctx context.Context) {
+	defer func() {
+		log.Info("notifyAndWaitExec finished")
```
Comment (liuzix) [inline cdc/sink/mysql/mysql.go:447]: Will remove after PR is stabilized

File: cdc/sink/mysql/mysql.go [hunk 2]
```diff
@@ -453,7 +468,9 @@ func (s *mysqlSink) notifyAndWaitExec(ctx context.Context) {
 	// scenario happens, the blocked goroutine will be leak.
 	select {
 	case <-ctx.Done():
+		return
```
Comment (liuzix) [inline cdc/sink/mysql/mysql.go:471]: will remove

File: cdc/sink/mysql/mysql.go [hunk 3]
```diff
@@ -453,7 +468,9 @@ func (s *mysqlSink) notifyAndWaitExec(ctx context.Context) {
...
 	case <-done:
+		return
```
Comment (liuzix) [inline cdc/sink/mysql/mysql.go:473]: ditto

**Review by sdojjy — APPROVED** [review]

File: cdc/sink/mysql/mysql.go (hunk 1 above)
Comment (asddongmen) [reply cdc/sink/mysql/mysql.go:453]: Should you remove this now?

File: cdc/sink/mysql/mysql.go [hunk 4]
```diff
@@ -273,6 +286,11 @@ func (s *mysqlSink) flushRowChangedEvents(ctx context.Context) {
 		if len(resolvedTxnsMap) != 0 {
 			s.dispatchAndExecTxns(ctx, resolvedTxnsMap)
 		}
+		for _, worker := range s.workers {
```
Comment (asddongmen) [inline cdc/sink/mysql/mysql.go:289]: Could you add a comments to tell why we need to go to `outer` if a worker `!isNormal`?

**Review by asddongmen — APPROVED** [review]

**Pull request merged**